packages/store/test/test-rankOrder.js

```diff
@@ -22,6 +23,177 @@ const alice = Far('alice', {});
 const bob = Far('bob', {});
 const carol = Far('carol', {});
 
+/**
+ * A factory for arbitrary passables
+ */
+const { passable } = fc.letrec(tie => {
+  return {
+    passable: tie('dag').map(x => harden(x)),
+    dag: fc.oneof(
+      { depthFactor: 0.5, withCrossShrink: true },
+      // a tagged value whose payload is an array of [key, leaf] pairs
+      // where each key is unique within the payload
+      // XXX can the payload be generalized further?
+      fc
+        .record({
+          type: fc.constantFrom('copyMap', 'copySet', 'nonsense'),
+          payload: fc.set(fc.fullUnicodeString(), { maxLength: 3 }).chain(k => {
+            return fc.tuple(fc.constant(k), tie('leaf'));
+          }),
+        })
+        .map(({ type, payload }) => makeTagged(type, payload)),
+      fc.array(tie('dag'), { maxLength: 3 }),
+      fc.dictionary(
+        fc.fullUnicodeString().filter(s => s !== 'then'),
+        tie('dag'),
+        { maxKeys: 3 },
+      ),
+      tie('dag').map(v => Promise.resolve(v)),
+      tie('leaf'),
+    ),
+    leaf: fc.oneof(
+      fc.record({}),
+      fc.fullUnicodeString(),
+      fc.fullUnicodeString().map(s => Symbol.for(s)),
+      fc.fullUnicodeString().map(s => new Error(s)),
+      // primordial symbols and registered lookalikes
+      fc.constantFrom(
+        ...Object.getOwnPropertyNames(Symbol).flatMap(k => {
+          const v = Symbol[k];
+          if (typeof v !== 'symbol') return [];
+          return [v, Symbol.for(k), Symbol.for(`@@${k}`)];
+        }),
+      ),
+      fc.bigInt(),
+      fc.integer(),
+      fc.constantFrom(-0, NaN, Infinity, -Infinity),
+      fc.constantFrom(null, undefined, false, true),
+      fc.constantFrom(alice, bob, carol),
+      // unresolved promise
+      fc.constant(new Promise(() => {})),
+    ),
+  };
+});
+
+test('compareRank is reflexive', async t => {
+  await fc.assert(
+    fc.property(passable, x => {
+      return assertionPassed(
+        t.is(compareRank(x, x), 0),
+        () => compareRank(x, x) === 0,
+      );
+    }),
+  );
+});
+
+test('compareRank totally orders ranks', async t => {
+  await fc.assert(
+    fc.property(passable, passable, (a, b) => {
+      const ab = compareRank(a, b);
+      const ba = compareRank(b, a);
+      if (ab === 0) {
+        return assertionPassed(t.is(ba, 0), () => ba === 0);
+      }
+      return assertionPassed(t.is(Math.sign(ba), -Math.sign(ab)), () => {
+        return Object.is(Math.sign(ba), -Math.sign(ab));
```

Comment:
If `ab` and `ba` are both `NaN`, this will pass but should not. This is a realistic edge case because our partial order comparisons like `compareKeys` return `NaN` to indicate incomparable.

Reply:
Isn't `compareRank` documented in packages/store/src/types.js to only return -1, 0, or 1?

Reply:
Correct. But I'd still prefer the test to be robust against that case.